modules/blockexplorer/hashes.go

```diff
@@ -0,0 +1,259 @@
+package blockexplorer
+
+import (
+	"bytes"
+	"errors"
+
+	"github.com/NebulousLabs/Sia/crypto"
+	"github.com/NebulousLabs/Sia/encoding"
+	"github.com/NebulousLabs/Sia/types"
+)
+
+const (
+	responseBlock        = "Block"
+	responseTransaction  = "Transaction"
+	responseFileContract = "FileContract"
+	responseOutput       = "Output"
+	responseAddress      = "Address"
+)
+
+type (
+	// Many explicit wrappers for return values
+	// All must have the ResponseType field as a string
+	blockResponse struct {
+		Block        types.Block
+		Height       types.BlockHeight
+		ResponseType string
+	}
+
+	// Wrapper for a transaction, with a little extra info
+	transactionResponse struct {
+		Tx           types.Transaction
+		ParentID     types.BlockID
+		TxNum        int
+		ResponseType string
+	}
+
+	// Wrapper for fcInfo struct, defined in database.go
+	fcResponse struct {
+		Contract     crypto.Hash
+		Revisions    []crypto.Hash
+		Proof        crypto.Hash
+		ResponseType string
+	}
+
+	// Wrapper for the address type response
+	addrResponse struct {
+		Txns         []crypto.Hash
+		ResponseType string
+	}
+
+	outputResponse struct {
+		OutputTx     crypto.Hash
+		InputTx      crypto.Hash
+		ResponseType string
+	}
+)
+
+// GetHashInfo returns sufficient data about the hash that was
+// provided to do more extensive lookups
+func (be *BlockExplorer) GetHashInfo(hash []byte) (interface{}, error) {
+	if len(hash) < crypto.HashSize {
+		return nil, errors.New("requested hash not long enough")
+	}
+
+	lockID := be.mu.RLock()
+	defer be.mu.RUnlock(lockID)
+
+	// Perform a lookup to tell which type of hash it is
+	typeBytes, err := be.db.GetFromBucket("Hashes", hash[:crypto.HashSize])
+	if err != nil {
+		return nil, err
+	}
+	if typeBytes == nil {
+		return nil, errors.New("requested hash not found in database")
+	}
```

Comment:
shouldn't this condition be checked for (and returned as an error) in the `GetFromBucket` function? Or are there cases where we want to check for nil explicitly (i.e. to branch on it)?